datafusion/sqllogictest/bin/sqllogictests.rs

```diff
@@ -40,19 +40,25 @@ pub fn main() -> Result<()> {
         .block_on(run_tests())
 }
 
-fn value_validator(actual: &[Vec<String>], expected: &[String]) -> bool {
-    let expected = expected
-        .iter()
-        // Trailing whitespace from lines in SLT will typically be removed, but do not fail if it is not
-        // If particular test wants to cover trailing whitespace on a value,
-        // it should project additional non-whitespace column on the right.
-        .map(|s| s.trim_end().to_owned())
-        .collect::<Vec<_>>();
+#[allow(clippy::ptr_arg)]
+fn normalizer(s: &String) -> String {
+    // Trailing whitespace from lines in SLT will typically be removed, but do not fail if it is not
+    // If particular test wants to cover trailing whitespace on a value,
+    // it should project additional non-whitespace column on the right.
+    s.trim_end().to_owned()
+}
+
+fn value_validator(
+    normalizer: Normalizer,
```

Comment:
Not related to this PR, I'm a bit confused about why sqllogictests need a `Normalizer`🤔 
The value validator can already be customized, and specifically, the validator will call the normalizer within its proces. `Normalizer` seems like an unnecessary design to me.

Reply:
> Not related to this PR, I'm a bit confused about why sqllogictests need a `Normalizer`🤔 The value validator can already be customized, and specifically, the validator will call the normalizer within its proces. `Normalizer` seems like an unnecessary design to me.

That was not entirely true unfortunately, the normalization was hardcoded in places.

Reply:
> That was not entirely true unfortunately, the normalization was hardcoded in places.

`normalize_string` was an internal part of the default validator.  Although it was hard-coded, the outer validator can be user-defined. A user-defined validator like the one provided by DataFusion can offer more flexibility to meet specific row comparison needs, and might not require normalization.

Reply:
Yes. However, in my sidebranch for the sqlite tests it made things a bit cleaner, and in the branch I used to generate the 'cleansed' .slt files it was necessary as normalization was required where validation was not. I have no desire to spend any more time working on that dependency tbh so if you feel strongly about it I'll point you at that project :)

Reply:
Thank you @Omega359 for the explanation. I don't have a strong opinion on this, just curious about its use cases. Since it can make some things cleaner, its existence is reasonable.